Dockerfile

```diff
@@ -0,0 +1,81 @@
+FROM hexpm/elixir:1.15.7-erlang-26.1-debian-bookworm-20230612-slim as base
+
+# install build dependencies
+# --allow-releaseinfo-change allows to pull from 'oldstable'
+RUN apt-get update --allow-releaseinfo-change -y && \
+    apt-get install -y \
+    build-essential \
+    git \
+    openssl \
+    ca-certificates \
+    inotify-tools && \
+    apt-get clean && \
+    rm -f /var/lib/apt/lists/*_*
+
+# Install hex
+RUN mix local.hex --force && \
+    mix local.rebar --force && \
+    mix hex.info
+
+WORKDIR /src
+
+FROM base as deps
+# Cache elixir deps
+ADD mix.exs mix.lock ./
+RUN mix do deps.get, deps.compile
+
+FROM deps as builder
+# Add all the rest
+ADD . .
```

Comment:
If everything is being added, could this stage be skipped?

Reply:
The layer explicitly separates the compilation of dependencies and the build of the code, to prevent the `rebuild` from taking a long time. You could also manage without the layer, but it seems clearer this way - by porting from the previous version I just lost this information -

Reply:
Ah, now I get it!
Then it's ok for me. I would suggest to make clearer however which sections of this dockerfile are common, which ones are only for `dev` (if any) and which ones are only for `prod`. The `#----` separator alone looks still a bit obscure

Reply:
After discussing it with @noaccOS, I've just added a comment as well